### DISCUSSION TITLE
Welcome to commonroad-scenario-designer Discussions!

### DISCUSSION THREAD

**Discussion by smaierhofer** [CommonRoad/commonroad-scenario-designer#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ataparlar** [comment]
Hi,
I am Ata Parlar. I am currently working on mapping for autonomous vehicles. Currently, working for Leo Drive and Autoware Foundation. I used CommonRoad Sceneraio Designer GUI and Python API for converting different formatted HD-Maps to Lanelet2 Maps. I would like to contribute to development the tool.

I didn't understand that what is the number in the starting of the branch names? Are they the issue numbers in CommonRoad forum or somewhere else? I could not match any of them with the branches so, I needed to ask. 

Thank you for your support.
Ata Parlar

**Reply by smaierhofer** [reply]
These are issue numbers from our internal Gitlab repository. This repository is mirrored from there (and back).

**Comment by oliverpapst1** [comment]
Hello all,
I am Oliver and I am working in the area of automated driving. Currently I am using the CR converter to try to include sidewalk information contained in .osm files into lanelet2 files.

I tried using the .osm files and conversion to lanelet2 scripts from the repository, but as of now it seems to me that sidewalk information gets
lost during the intermediate conversion from .osm to the CR XML scheme.
Any help regarding this matter would be highly appreciated.

Thanks a lot in advance.
Yours, Oliver

**Reply by smaierhofer** [reply]
Hi Oliver,

we added few years ago an experimental feature for that called sublayer extraction.
Unfortunately, we didn't have the capacity to further improve and maintain it.
I just tested it and it doesn't work with the current default OSM parameters. 
These are the relevant config parameters:
- [EXTRACT_SUBLAYER](https://github.com/CommonRoad/commonroad-scenario-designer/blob/0ae7e126f0307246c38f0afc1d4f7326438670cf/crdesigner/common/config/osm_config.py#L74C5-L74C21)
- [ACCEPTED_HIGHWAYS_SUBLAYER](https://github.com/CommonRoad/commonroad-scenario-designer/blob/0ae7e126f0307246c38f0afc1d4f7326438670cf/crdesigner/common/config/osm_config.py#L77)

Maybe other OSM parameters need to be changed to get it working again.

Best,
Sebastian

**Reply by oliverpapst1** [reply]
Hi Sebastian,

thank you for your response, I will try these config parameters and see whether I will be able to extract the sidewalk information during the .osm -> CR conversion step, it seems to get lost there as some CR -> LL2 examples work with your repository.

Best,
Oliver

**Reply by oliverpapst1** [reply]
Hi Sebastian,

there are some examples of .xml files in CR format containing lanelet type sidewalk tags e.g.

`<laneletType>sidewalk</laneletType>`

e.g. in the file traffic_priority_lanelets_utm.xml

do you by any chance know where this type is set in your code?

Best,
Oliver

**Reply by smaierhofer** [reply]
I guess it is extracted from the OSM edge types. For sublayers the sidewalk is manually set [here](https://github.com/CommonRoad/commonroad-scenario-designer/blob/06a969934b9eedf1b3455f12badae7ee456156cd/crdesigner/map_conversion/osm2cr/converter_modules/intermediate_operations/intermediate_format/_intermediate_format.py#L453).

**Reply by oliverpapst1** [reply]
Thank you for your fast response and link to the code file, I will check that now.

Just one more question in the meantime, why is the SUBLAYER_LANELETTYPE defined as sidewalk i.e. why is there only one possible type for a SUBLAYER LANELET?

`SUBLAYER_LANELETTYPE = Attribute('sidewalk', "Sublayer Lanelet Type")`

**Reply by smaierhofer** [reply]
Unfortunately, I can't remember the reason. Maybe it is possible to add several types (I'm not sure without debugging). 

btw., crossing lanelets are handled separately, see [here](https://github.com/CommonRoad/commonroad-scenario-designer/blob/06a969934b9eedf1b3455f12badae7ee456156cd/crdesigner/map_conversion/osm2cr/converter_modules/intermediate_operations/intermediate_format/_intermediate_format.py#L487).

**Reply by oliverpapst1** [reply]
No problem, thanks a lot for the update and information regarding crossings.

**Reply by oliverpapst1** [reply]
[merge function](https://github.com/CommonRoad/commonroad-scenario-designer/blob/06a969934b9eedf1b3455f12badae7ee456156cd/crdesigner/map_conversion/osm2cr/converter_modules/intermediate_operations/intermediate_format/_intermediate_format.py#L453)

Hi Sebastian,

I wanted to ask you, what is meant with "other instance" in the docstring of the merge() function where the sidewalk information is added?

Best,
Oliver

**Reply by smaierhofer** [reply]
It is the road network for vehicles

**Reply by OliverPapst** [reply]
In other words - solely the roads a vehicle is supposed to drive on?

**Reply by smaierhofer** [reply]
yes

**Reply by oliverpapst1** [reply]
Thank you!

Do I get that right that this merge function is solely responsible for merging sidewalks into the road network?
As all edge types are set to sidewalk in the merge function.

**Reply by smaierhofer** [reply]
yes, it should merge the sidewalk and vehicle roads

**Reply by oliverpapst1** [reply]
thank you!

I also noticed you set LANEWIDTHS in the osm_config.py, can you please let me know where the lane width information comes from for the different types?
e.g.: "trunk_link": 3.5

**Reply by smaierhofer** [reply]
It is an approximate value based on an online research.
